edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/WeeklyAvaliability.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Runtime.InteropServices.ComTypes;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class WeeklyAvaliability
+    {
+        private readonly ISystemTime time;
+
+        public string Name;
+        public string Version;
+
+        private Avaliability[] dailyAvaliabilities;
+        private DayOfWeek currentDay;
+
+        public WeeklyAvaliability(string name, string version, ISystemTime time)
+        {
+            this.Name = name;
+            this.Version = version;
+            this.time = time;
+
+            this.currentDay = time.UtcNow.DayOfWeek;
+            this.dailyAvaliabilities = Enumerable.Range(0, 7).Select(_ => new Avaliability(name, version, time)).ToArray();
+        }
+
+        public WeeklyAvaliability(WeeklyAvaliabilityRaw raw, ISystemTime time)
+        {
+            this.Name = raw.Name;
+            this.Version = raw.Version;
+            this.time = time;
+
+            this.currentDay = time.UtcNow.DayOfWeek;
+            this.dailyAvaliabilities = raw.DailyAvaliabilities.Select(aRaw => new Avaliability(aRaw, time)).ToArray();
+
+            if (time.UtcNow.DayOfWeek != raw.SavedDate.DayOfWeek)
+            {
+                int difference = (int)(time.UtcNow.Date - raw.SavedDate).Days;
```

Comment:
Days property should be int type
```suggestion
                int difference = (time.UtcNow.Date - raw.SavedDate).Days;
```